qsirecon/cli/parser.py

```diff
@@ -442,39 +447,84 @@ def parse_args(args=None, namespace=None):
             f"total threads (--nthreads/--n_cpus={config.nipype.nprocs})"
         )
 
-    bids_dir = config.execution.bids_dir
+    input_dir = config.execution.input_dir
     output_dir = config.execution.output_dir
     work_dir = config.execution.work_dir
     version = config.environment.version
 
-    # Update the config with an empty dict to trigger initialization of all config
-    # sections (we used `init=False` above).
-    # This must be done after cleaning the work directory, or we could delete an
-    # open SQLite database
-    config.from_dict({})
-
     # Ensure input and output folders are not the same
-    if output_dir == bids_dir:
+    if output_dir == input_dir:
         parser.error(
             "The selected output folder is the same as the input BIDS folder. "
             "Please modify the output path (suggestion: %s)."
-            % bids_dir
+            % input_dir
             / "derivatives"
             / ("qsirecon-%s" % version.split("+")[0])
         )
 
-    if bids_dir in work_dir.parents:
+    if input_dir in work_dir.parents:
         parser.error(
             "The selected working directory is a subdirectory of the input BIDS folder. "
             "Please modify the output path."
         )
 
     # Setup directories
-    config.execution.log_dir = config.execution.output_dir / "logs"
+    log_dir = output_dir / "logs"
     # Check and create output and working directories
-    config.execution.log_dir.mkdir(exist_ok=True, parents=True)
+    log_dir.mkdir(exist_ok=True, parents=True)
     work_dir.mkdir(exist_ok=True, parents=True)
 
+    # Run ingression if necessary
+    if config.workflow.input_type in ("hcpya", "ukb"):
+        import os.path as op
+        import shutil
+
+        from ingress2qsirecon.data import load_resource
+        from ingress2qsirecon.utils.functions import create_layout
+        from ingress2qsirecon.utils.workflows import create_ingress2qsirecon_wf
+
+        # Fake BIDS directory to be created
+        config.execution.bids_dir = work_dir / "bids"
+
+        # Make fake BIDS files
+        bids_scaffold = str(load_resource("bids_scaffold"))
+        if not op.exists(op.join(config.execution.bids_dir, "dataset_description.json")):
```

Comment:
could you switch this to use pathlib instead of op? I think we're going to try to consistently use it to be in line with nipreps

Reply:
Think I changed this, untested so far:
```
if config.workflow.input_type in ("hcpya", "ukb"):
        import shutil

        from ingress2qsirecon.data import load_resource
        from ingress2qsirecon.utils.functions import create_layout
        from ingress2qsirecon.utils.workflows import create_ingress2qsirecon_wf

        # Fake BIDS directory to be created
        config.execution.bids_dir = work_dir / "bids"

        # Make fake BIDS files
        bids_scaffold = load_resource("bids_scaffold")
        if not (config.execution.bids_dir / "dataset_description.json").exists():
            shutil.copytree(
                bids_scaffold,
                config.execution.bids_dir,
                dirs_exist_ok=True,
            )
```